packages/drivers/routerlicious-driver/src/documentServiceFactory.ts

```diff
@@ -103,7 +105,8 @@ export class RouterliciousDocumentServiceFactory implements IDocumentServiceFact
             throw new Error(`Token was not provided.`);
```

Comment:
This code above no longer looks right - I'd remove requirement to pass token and always rely (and require) on token provider. Otherwise you are asking for both things - token & token provider. That's too much, but also not clear what token is it - storage or socket? That also results in telemetry (that you will need to add around all token fetches) to be spread out instead of being in one place, etc.

Reply:
Good catch! I am providing either options for now but forgot to remove the error throwing part. 

Currently, Either you provide a token as part of resolved url or you provide a token provider. For the former, I am creating a default provider based on your provided token.

So I added two different token types in case the host wants that. But currently for r11s/FRS, there is just one token. The right answer will become clear as we move forward. But wanted to keep the interface generic for now.